logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/Conversation.kt

```diff
@@ -51,6 +49,71 @@ data class Conversation(
     val receiptMode: ReceiptMode,
 ) {
 
+    companion object {
+        /**
+         * A set of default [AccessRole] valid for Group Conversations
+         * for both personal and team users.
+         *
+         * @see [AccessRole]
+         */
+        val defaultGroupAccessRoles = setOf(AccessRole.TEAM_MEMBER, AccessRole.NON_TEAM_MEMBER)
+
+        /**
+         * A set of default [Access] modes valid for Group Conversations
+         * for both personal and team users.
+         *
+         * @see [Access]
+         */
+        val defaultGroupAccess = setOf(Access.INVITE)
+
+        /**
+         * Returns a sensible set of [AccessRole] given a combination of
+         * flags
+         *
+         * @see [AccessRole]
+         */
+        fun accessRolesFor(
+            guestAllowed: Boolean,
+            servicesAllowed: Boolean,
+            nonTeamMembersAllowed: Boolean
+        ): Set<AccessRole> =
+            defaultGroupAccessRoles.toMutableSet().apply {
+                if (servicesAllowed) {
+                    add(AccessRole.SERVICE)
+                } else {
+                    remove(AccessRole.SERVICE)
+                }
+
+                if (guestAllowed) {
+                    add(AccessRole.GUEST)
+                } else {
+                    remove(AccessRole.GUEST)
+                }
+
+                if (nonTeamMembersAllowed) {
+                    add(AccessRole.NON_TEAM_MEMBER)
+                } else {
+                    remove(AccessRole.NON_TEAM_MEMBER)
```

Comment:
what's the point of all this `remove(AccessRole...)` if we know that it's not there already. 
can be just like 
```
            defaultGroupAccessRoles.toMutableSet().apply {
                if (servicesAllowed) add(AccessRole.SERVICE)

                if (guestAllowed) add(AccessRole.GUEST)

                if (nonTeamMembersAllowed) add(AccessRole.NON_TEAM_MEMBER)
           }
```